### PR TITLE
Program list on homepage: use semantic HTML, improve element order for screen readers

### LINKED ISSUE #1688: on Home page, program cards missing heading
> Wrap the currently offered programs (and the "more" placeholder) in a h4 tag.

### PR DESCRIPTION
**Pull request by singingwolfboy**
#### What are the relevant tickets?
Fixes #1688.

#### What's this PR do?
Makes the program list on the program home page more accessible, particularly to screen readers.

#### Where should the reviewer start?
Examine the HTML to make sure the structure makes sense. Load the home page in your browser to make sure it looks good visually.

#### How should this be manually tested?
I recommend installing a screen reader like [ChromeVox](http://www.chromevox.com/) to experience the page the way a non-sighted user would, and make sure that going through the page element by element makes sense.

### CODE REVIEW

**Comment by pdpinch** [comment]
nitpick: can you update the title to be more specific to what you're actually changing?

**Comment by noisecapella** [comment]
@singingwolfboy FYI I rebased on master to fix a PR deployment issue

File: cms/templates/cms/home_page.html [hunk 1]
```diff
@@ -118,79 +118,74 @@ <h5 class="upper-case">Save Money &amp; Time</h5>
...
+                       alt="course image for {{ program }}" class="program-default-image">
                 {% endif %}
-                <div class="program-thumbnail-description">
-                  <p>{{ program.description|default:"No description available for this program." }}</p>
+              {% if program.programpage %}
+                </a>
+              {% endif %}
+            </div>
+            <div class="program-description">
+              <p id="program-{{ program.id }}-description">{{ program.description|default:"No description available for this program." }}</p>
```
Comment (noisecapella) [inline cms/templates/cms/home_page.html]: What is this `id` for?

**Comment by noisecapella** [comment]
Just some questions, functionality looks good on desktop and mobile

File: cms/templates/cms/home_page.html (hunk 1 above)
Comment (singingwolfboy) [inline cms/templates/cms/home_page.html]: The `<h4>` element for the program title has an [`aria-describedby`](http://rawgit.com/w3c/aria/master/aria/aria.html#aria-describedby) attribute that references this ID. It allows the screen reader to read the program description immediately after reading the program title.

File: cms/templates/cms/home_page.html [hunk 2]
```diff
@@ -118,79 +118,74 @@ <h5 class="upper-case">Save Money &amp; Time</h5>
...
-                  {% if program.programpage.external_program_page_url %}
-                    <a href="{{ program.programpage.external_program_page_url }}" class="program-link">
-                  {% else %}
-                    <a href="{{ program.programpage.url }}" class="program-link">
-                  {% endif %}
+      <section class="row current-programs-section">
+        <h3>Current MITx MicroMasters Programs</h3>
+        <ul class="current-programs-list">
+        {% for program in programs %}
+          <li class="program-card col-md-4">
```
Comment (singingwolfboy) [inline cms/templates/cms/home_page.html]: I thought I was going to use it for styling, and then it turned out that I didn't have to. I've removed it from this pull request.

**Pull request merged**